src/Data/Digit.agda

```diff
@@ -107,9 +107,8 @@ toDigits base@(suc (suc k)) n = <′-rec Pred helper n
 
   helper : ∀ n → <′-Rec Pred n → Pred n
   helper n                       rec with n divMod base
-  helper .(toℕ r + 0     * base) rec | result zero    r refl = ([ r ] , refl)
-  helper .(toℕ r + suc x * base) rec | result (suc x) r refl =
-    cons r (rec (suc x) (lem x k (toℕ r)))
+  ... | result zero    r eq = ([ r ] , P.sym eq)
+  ... | result (suc x) r refl = cons r (rec (lem x k (toℕ r)))
```

Comment:
This was an only-partially-successful attempt to make `helper` more lazy in its `with`-bound argument `eq`. Can be omitted from this PR in favour of a more thorough consideration later...

Reply:
Looks like an improvement to me, so happy to keep it in :+1: